eahub/localgroups/forms.py

```diff
@@ -1,45 +1,94 @@
 from django import forms
+from django.db import models
 
-from . import models
+from . import models as localgroups_models
 from ..base import models as base_models
 from ..profiles import models as profiles_models
 
 
-def unwrap_user(value):
-    if isinstance(value, base_models.User):
-        return value.profile
-    return value
-
-
 class UserMultipleChoiceField(forms.ModelMultipleChoiceField):
-    def __init__(self, **kwargs):
-        forms.ModelMultipleChoiceField.__init__(
-            self,
-            queryset=profiles_models.Profile.objects.all(),
-            to_field_name="slug",
-            **kwargs,
+    def __init__(self, *, user, local_group, **kwargs):
+        if local_group is None or local_group.pk is None:
+            already_selected = models.Value(False, output_field=models.BooleanField())
+        else:
+            already_selected = models.Case(
+                models.When(pk__in=local_group.organisers.all(), then=True),
+                default=False,
+                output_field=models.BooleanField(),
+            )
+        queryset = base_models.User.objects.select_related("profile").annotate(
+            already_selected=already_selected
```

Comment:
I assume if a user is annotated as already selected, this does not block them from being selected as organiser for another group? Don't fully comprehend the code, so just want to check that's the case. I assume we want users to be able to be organisers of different groups

Reply:
`already_selected` is an annotation that only exists within a particular query, not a persistent database field. This query is only concerned with the particular group that's being edited and is completely independent of whatever's going on in any other groups.